### PR TITLE
JavaScript - Number Recognizers for Portuguese

### PR DESCRIPTION
**Pull request by lucasmarambio**
Ported all Number extractors and parsers (Portuguese) from C# to TypeScript/JavaScript implementation

### CODE REVIEW

File: JavaScript/recognizers-number/src/number/portuguese/parserConfiguration.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                result.push(tempWord);
+                return;
+            }
+
+            // ends with 'avo' or 'ava'
+            if (PortugueseNumeric.WrittenFractionSuffix.some(suffix => tempWord.endsWith(suffix))) {
+                let origTempWord = tempWord;
+                let newLength = origTempWord.length;
+                tempWord = origTempWord.substring(0, newLength - 3);
+                if (!tempWord){
```
Comment (tellarin) [inline JavaScript/recognizers-number/src/number/portuguese/parserConfiguration.ts]: Nitpick: whitespace before {

File: JavaScript/recognizers-number/src/number/portuguese/parserConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        let finalValue = 0;
+        let strBuilder = "";
+        let lastGoodChar = 0;
+        for (let i = 0; i < numberStr.length; i++) {
+            strBuilder = strBuilder.concat(numberStr[i]);
+            if (this.cardinalNumberMap.has(strBuilder) && this.cardinalNumberMap.get(strBuilder) > value) {
+                lastGoodChar = i;
+                value = this.cardinalNumberMap.get(strBuilder);
+            }
+            if ((i + 1) === numberStr.length) {
```
Comment (tellarin) [inline JavaScript/recognizers-number/src/number/portuguese/parserConfiguration.ts]: Nitpick: Blank lines between if statements.

File: Patterns/Portuguese/Portuguese-Numbers.yaml [hunk 3]
```diff
@@ -31,9 +31,6 @@ AllIntRegex: !nestedRegex
 AllPointRegex: !nestedRegex
   def: ((\s+{ZeroToNineIntegerRegex})+|(\s+{AllIntRegex}))
   references: [ ZeroToNineIntegerRegex, AllIntRegex ]
-AllFloatRegex: !nestedRegex
```
Comment (tellarin) [inline Patterns/Portuguese/Portuguese-Numbers.yaml]: Why was this removed from the YAML file? Shouldn't it just be moved up?

Comment (lucasmarambio) [reply Patterns/Portuguese/Portuguese-Numbers.yaml]: This Regex was removed because it was duplicated. In the generated NumbersDefinitions for C# this line didn't present any problem because the duplicated regex is ignored, but for Javascript it was neccesary to delete it.

File: JavaScript/recognizers-number/src/number/portuguese/parserConfiguration.ts (hunk 2 above)
Comment (lucasmarambio) [reply JavaScript/recognizers-number/src/number/portuguese/parserConfiguration.ts]: Done!

File: JavaScript/recognizers-number/src/number/portuguese/parserConfiguration.ts (hunk 1 above)
Comment (lucasmarambio) [reply JavaScript/recognizers-number/src/number/portuguese/parserConfiguration.ts]: Done!

File: Patterns/Portuguese/Portuguese-Numbers.yaml (hunk 3 above)
Comment (tellarin) [reply Patterns/Portuguese/Portuguese-Numbers.yaml]: Sorry, I didn't realize that.
